internal/pkg/suite/steps.go

```diff
@@ -0,0 +1,231 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package suite
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"fmt"
+	"github.com/cucumber/messages-go/v10"
+	"github.com/kubernetes-sigs/ingress-controller-conformance/internal/pkg/k8s"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"time"
+)
+
+func iHaveAnIngressNamedInTheNamespace(ingressName string, namespace string) error {
```

Comment:
A nit;

I think it's important to keep the regex and the handler close together. AFAICT there's no good way in the `godog` API to do that, so maybe we can splat all this into init functions.

```Go
package suite

var DefaultSuite := godoc.Suite{}

func init() {
        iHaveAnIngressNamedInTheNamespace()
}

func iHaveAnIngressNamedInTheNamespace() {
        DefaultSuite.Step(
                `^I have an Ingress named "([^"]*)" in the "([^"]*)" namespace$`,
                func (ingressName string, namespace string) error {
                        return nil
                }
        )
}

```

Or maybe:
```Go
package suite

func RegisterStep(expr interface{}, stepFunc interface{}) error {
        DefaultSuite.Step(expr, stepFunc)
        return nil
}

var _ =   RegisterStep(
        `^I have an Ingress named "([^"]*)" in the "([^"]*)" namespace$`,
        func (ingressName string, namespace string) error {
                return nil
        }
)
```

Reply:
Actually, maybe you can't do this because I don't think you are supposed to create your own `*godog.Suite`?